src/Raven.Client/Documents/DocumentStore.cs

```diff
@@ -249,6 +247,27 @@ public override IDocumentStore Initialize()
             return this;
         }
 
+        private void OnRequestExecutorCreated(object sender, RequestExecutor executor)
```

Comment:
This is somewhat of a problem. 
The issue is that if we define this per db, then we could have different values, but then things like `OpenSession(db)` come into play